Added fix for token id issue

cryptonoyaiba opened the pull request:
- Added fix for token id issue and now no longer uses total supply to assign it
- Added test in suite to check the logic of new token id generation
- Ran slither
- locked package versions
- added .deps to gitignore

cryptonoyaiba committed:
Added fix for token id issue

- Added fix for token id issue and now no longer uses total supply to assign it
- Added test in suite to check the logic of new token id generation
- Ran slither
- locked package versions
- added .deps to gitignore

ch4r10t33r reviewed (approved):
LGTM Please deploy the contracts on testnet and let me know the contract addresses.

vignesha22 reviewed (approved):
LGTM

cryptonoyaiba committed:
Added Goerli to hardhat config and modified deployment script

- Added Goerli network to configs
- Added deployment script for MembershipNFT contract
- Added script inside deployment that will verify contract on Etherscan
